packages/cells/src/widget.ts

```diff
@@ -253,13 +269,37 @@ class Cell extends Widget {
   }
 
   /**
-   * The view state of input being collapsed.
+   * The view state of input being hidden.
    */
-  get inputCollapsed(): boolean {
-    return this._inputCollapsed;
+  get inputHidden(): boolean {
+    return this._inputHidden;
   }
-  set inputCollapsed(value: boolean) {
-    this._inputCollapsed = value;
+  set inputHidden(value: boolean) {
+    if (this._inputHidden === value) {
+      return;
+    }
+    let layout = this._inputWrapper.layout as PanelLayout;
+    if (value) {
+      layout.removeWidget(this._input);
```

Comment:
I think it is important in this case to not have the elements in the DOM, switching to setting `parent` to `null`.